### PR TITLE
gickup: init at 0.10.18

### PR DESCRIPTION
**Pull request by rhousand**
## Things done

- Built on platform(s)
  - [ x ] x86_64-linux
  - [ ] aarch64-linux
  - [ x ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ x ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Result of `nixpkgs-review` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gickup</li>
  </ul>
</details>

### CODE REVIEW

File: pkgs/tools/backup/gickup/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+buildGoModule rec {
+  pname = "gickup";
+  version = "0.10.17";
+
+  src = fetchFromGitHub {
+    owner = "cooperspencer";
+    repo = "gickup";
+    rev = "v${version}";
+    sha256 = "sha256-tiQmb7bBWb99k23lS+d+YR14y4YeYPWqccl/2DLv7Dk=";
```
Comment (fabaff) [inline pkgs/tools/backup/gickup/default.nix:14]: Nowadays, using the SRI hash is the default.

```suggestion
    hash = "sha256-tiQmb7bBWb99k23lS+d+YR14y4YeYPWqccl/2DLv7Dk=";
```

File: pkgs/tools/backup/gickup/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  };
+
+  vendorSha256 = "sha256-DWGrs/ZKMKgVfwU7W+dktLELbW9Co7cmDy9pWVP5p2w=";
```
Comment (fabaff) [inline pkgs/tools/backup/gickup/default.nix:17]:
```suggestion
  vendorHash = "sha256-DWGrs/ZKMKgVfwU7W+dktLELbW9Co7cmDy9pWVP5p2w=";
```

File: pkgs/tools/backup/gickup/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description = "Backup all your repositories with Ease.";
```
Comment (fabaff) [inline pkgs/tools/backup/gickup/default.nix:20]:
```suggestion
    description = "Tool to backup repositories";
```

No period at the end (was mentioned in the past in CONTRIBUTING.md), the rest is opinionated.

File: pkgs/tools/backup/gickup/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    homepage = "https://github.com/cooperspencer/gickup";
```
Comment (fabaff) [inline pkgs/tools/backup/gickup/default.nix:21]: Tools (e.g., `nix-update`) could pick up the changelog data and add it to the commit messages and the PR description for package updates.

```suggestion
    homepage = "https://github.com/cooperspencer/gickup";
    changelog = "https://github.com/cooperspencer/gickup/releases/tag/v${version}";
```

**Review by adamcstephens — CHANGES_REQUESTED** [review]
output binary has no version, please add `ldflags = ["-X main.version=${version}"];`

https://nixos.org/manual/nixpkgs/stable/#ssec-go-common-attributes

**Comment by adamcstephens** [comment]
Please also rebase/rewrite your commit history to two commits. Adding yourself as maintainer, and init at 0.10.18

**Comment by adamcstephens** [comment]
0.10.20 is out, and the feedback suggested should be addressed.

**Pull request closed without merging**